common.md

```diff
@@ -28,18 +28,22 @@ to contracts with entrypoints. We may fix this issue in a later release.
   syntax TypeName ::= NullaryTypeName
                     | UnaryTypeName TypeName
                     | BinaryTypeName TypeName TypeName
+                    | BinaryPlusTypeName TypeName TypeName
```

Comment:
Should this not be declared as follows?

```suggestion
                    | BinaryPlusTypeName TypeName TypeNameList
```

Reply:
This is a good question, but the answer is actually that I don't want to do this because:

-   `Type` is the syntactic category of types used in the "outer" syntax (that is, the syntax a Michelson developer uses)
-   `TypeName` is the syntactic category of types used in the "inner" syntax (that is, how we refer to types internally)

Now, when I convert `Type`s to `TypeName`s, I add a normalization step. Part of that normalization is to pairs using the plus syntax into standard nested pairs. This way, we avoid having to change our internals based on the new `pair` plus syntax. The relevant rules are lines 38-39 of `common.md`.

Reply:
I see, thanks for the clarification.